src/qtt/algorithms/allxy.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Fri Sep 20 21:04:41 2019
+
+@author: eendebakpt
+"""
+
+from typing import Dict, Any, List, Union, Optional
+import numpy as np
+import matplotlib.pyplot as plt
+from lmfit import Model
+
+import qcodes
+from qtt.utilities.visualization import plot_vertical_line
+
+
+def generate_allxy_combinations() -> List[Any]:
+    """ Generate all combinations of the AllXY sequence from Reed 2013 """
+    xymapping = {'I': 'I', 'x': 'X90', 'y': 'Y90', 'X': 'X180', 'Y': 'Y180'}
+    allxy_combinations_input = ['II', 'XX', 'YY', 'XY', 'YX'] + ['xI', 'yI', 'xy', 'yx',
+                                                                 'xY', 'yX', 'Xy', 'Yx', 'xX', 'Xx', 'yY', 'Yy'] + ['XI', 'YI', 'xx', 'yy']
+    allxy_combinations = [(xymapping[x[0]], xymapping[x[1]]) for x in allxy_combinations_input]
+
+    return allxy_combinations
+
+
+def allxy_model(x: Union[float, np.ndarray], offset0: float, slope0: float, offset1: float, slope1: float, offset2: float, slope2: float) -> Union[float, np.ndarray]:
```

Comment:
All 6 parameters need to be individual arguments, otherwise the `lmfit` package will not work. The `x` is the index of the experiment, I will rename.
The type of `x` is such that you can apply the method both for a range of indices (e.g. an array) or a scalar (e.g. a float).